elastic/kibana
# [Uptime] [Alerts] update monitor status alerts to persist when documents are not found

## Summary

Fixes #99660 

Previously, alerts would auto resolve when we did not find down monitor documents. This created the opportunity for false negatives in alert state. By assuming that the absence of down documents meant up monitors, we would resolve the alert even if there weren't any up monitors at all. This happens in the case of the default monitor status alert, which has a timespan of 1 minute. When the monitor interval is greater than this default timespan, the query will at times return no documents at all, assume the alert is resolved, and resolve the alert.

This fixes this issue by checking for the presence of up monitors before resolving the alert. Because alert instances will resolve automatically if not fired in during the next scheduled action, we need a way to keep tract of the current alert instances. To that end `currentDownMonitors` as added to the alert state to reference in subsequent scheduled actions and to persist the current alert instance when no up monitors are found. 

To test:
 1. Create a consistently down monitor, like the 'always-down' monitor in the sample heartbeat config. Give that monitor an interval larger than the standard monitor status alert timespan, like 5 minutes.
 2. Add a default connector in the Settings page 
 3. On the overview page, toggle the status alert for the consistently down monitor to on.
 4. Observe that the monitor no longer resolves to UP if no documents are found within the timespan
 5. Change your heartbeat config so that your monitor comes up
 6. Observe that the alert resolves
